CMakeLists.txt

```diff
@@ -183,7 +184,8 @@ if(${TARGET} STREQUAL "all-test")
         create_symlinks moduleCkCache trace-converse moduleCommonLBs
         moduleTreeLB moduleCkMulticast moduleCkIO conv-cpm memory-os-wrapper
         threads-default-tls ldb-neighbor ldb-workstealing modulearmci
-        modulecollidecharm modulecollide memory-os memory-gnu-isomalloc)
+        modulecollidecharm modulecollide memory-os memory-gnu-isomalloc
+        ampi_globals_fs)
   if(CMK_CAN_LINK_FORTRAN)
```

Comment:
ampi_globals_fs is added below, `if(CMK_SUPPORTS_FSGLOBALS)`. Is that not working? LGTM otherwise.

Reply:
Hmm, I see. Then there must be something wrong with the fsglobals support. When I build on MacOS with `./build all-test netlrts-darwin-x86_64` (i.e., a non-parallel build) without this change, building all-test fails with:

```
../../../bin/ampicxx -g -o cxx-tlsglobals-shared-library-so.o -tlsglobals -Dtest_using_tlsglobals -Dprivatization_method=tlsglobals -Dtest_migration -Dtest_staticvars -Dtest_sharedlib -shared -c shared-library.C
../../../bin/ampicxx -g -o libcxx-tlsglobals-shared-library.so -tlsglobals -shared -standalone cxx-tlsglobals-shared-library-so.o
../../../bin/ampicxx -g -o cxx-tlsglobals -tlsglobals tlsglobals-test.o tlsglobals-test-cxx.o cxx-tlsglobals-other-object.o -balancer TreeLB -rpath-origin -L. -lcxx-tlsglobals-static-library -lcxx-tlsglobals-shared-library
../../../bin/ampicxx -g -o fsglobals-test.o -fsglobals -Dtest_using_fsglobals -Dprivatization_method=fsglobals -Dtest_staticvars -c test.C
../../../bin/ampicxx -g -o fsglobals-test-cxx.o -fsglobals -Dtest_using_fsglobals -Dprivatization_method=fsglobals -Dtest_staticvars -c test-cxx.C
../../../bin/ampicxx -g -o cxx-fsglobals-other-object.o -fsglobals -Dtest_using_fsglobals -Dprivatization_method=fsglobals -Dtest_staticvars -c other-object.C
../../../bin/ampicxx -g -o cxx-fsglobals-static-library.o -fsglobals -Dtest_using_fsglobals -Dprivatization_method=fsglobals -Dtest_staticvars -c static-library.C
ar cr libcxx-fsglobals-static-library.a cxx-fsglobals-static-library.o
ranlib libcxx-fsglobals-static-library.a
../../../bin/ampicxx -g -o cxx-fsglobals -fsglobals fsglobals-test.o fsglobals-test-cxx.o cxx-fsglobals-other-object.o -balancer TreeLB -rpath-origin -L. -lcxx-fsglobals-static-library
Fatal Error by charmc in directory /Users/mdiener/Work/charm5/netlrts-darwin-x86_64/tests/ampi/privatization
   ../../../bin/../lib/ampi_globals_fs.o: file not recognized: File truncated
charmc exiting...
make[6]: *** [cxx-fsglobals] Error 1
rm cxx-tlsglobals-static-library.o cxx-fsglobals-static-library.o cxx-tlsglobals-shared-library-so.o
make[5]: *** [build-privatization] Error 2
make[4]: *** [build-ampi] Error 2
make[3]: *** [all-test-tests] Error 2
make[2]: *** [CMakeFiles/all-test] Error 2
make[1]: *** [CMakeFiles/all-test.dir/all] Error 2
```

Reply:
Found the error: `CMK_SUPPORTS_FSGLOBALS` etc. is not yet defined at this point.

Reply:
Is there anything that can be done about that?

Reply:
I've put the corresponding check into AMPI's CMakeLists.txt for now.